### DISCUSSION TITLE
Comparing mutant vs normal cell types - Pseudobulk replicate generation

### DISCUSSION THREAD

**Discussion by michellemeier27** [GreenleafLab/ArchR#2035] [Questions / Documentation] [answered]
Hi there,

First of all, thanks for putting together such a comprehensive package, its been the only one that has worked for us to analyse our data! 
This discussion might be related to #696 , but I'd still love some clarification. 
The dataset I am working with contains 3 different cell types (VEC, LEC and AEC), and two different experimental conditions (mutant and wildtype). We are mainly interested in the differentially accessible peaks in VEC mutant cells vs VEC wild type cells (so only in one cell type). What I have done, is run the functions addGroupCoverages and addReproduciblePeakSet using the celltype (which corresponds to clusters) as groupBy variable and then ran getMarkerFeatures on a groupBy variable containing both celltype and condition information (so VEC_mutant vs VEC_WT). As the results made sense and were confirmed in the wet lab, I didn't give it any more thought. However, as I was cleaning up the code I started questioning whether it is appropriate to give a different groupBy variable for addGroupCoverages/addReproduciblePeakSet and getMarkerFeatures. Looking at the underlying assumption of generating pseudo-bulks (The underlying assumption in this process is that the single cells that are being grouped together are sufficiently similar that we do not care to understand the differences between them), I don't think this necessarily holds true as we do expect some form of difference between the mutant and wild type cells. However, the main source of variance in the dataset is definitively the cell types, not the condition (i.e. the clustering is driven by cell types). I have tried to re-run the analysis using a groupBy variable containing both levels of information similar to what you suggested in the issue mentioned above. The results I got from that seem very off. My guess for this is that is due to the very small number of cells per cluster. Unfortunately, the experimental set-up is very limiting when it comes to cell number, so I am trying to make the best out of a pretty messy situation. 

Is there an optimal strategy to run these types of analyses? Would the most sensible workflow be to subset the project to just the cell type we're interested in and then use the condition as groupBy variable? Would you consider the assumption for pseudo replicates met by saying the cell type identity is the highest source of variability? 

Happy to clarify things more. Thanks for your help, any discussion/idea/general comments would be greatly appreciated. 

Cheers,
Michelle

**Comment by rcorces** [comment, marked as answer]
There is no one-size-fits-all solution. In your analysis, you may miss a few peaks that are extremely specific to normal or mutant since you have grouped them together in the peak calling stage. but if you dont have enough cells to get good peak calls after splitting by normal and mutant, then I dont see any issue with doing what you've done. The ideal is to have enough cells. But in the absence of that, you have to work with what you have.

**Reply by michellemeier27** [reply]
Thanks for your quick reply, this was very helpful.
